### PR TITLE
(PIE-860) Write Readme

### PR DESCRIPTION
**Pull request by RandomNoun7**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,72 +1,291 @@
...
+5. [Installing A Processor](#installing-a-processor)
+6. [Token vs Username Password Authentication](#token-vs-username-password-authentication)
+7. [Resources Placed On The Machine](#resources-placed-on-the-machine)
+8. [Logging](#logging)
+9. [Advanced Configuration Options](#advanced-configuration-options)
 
 ## Intro
 
-The common events library is a module makes it easier to gather events from [Puppet API end points](https://puppet.com/docs/pe/2021.2/orchestrator_api_jobs_endpoint.html#orchestrator_api_jobs_endpoint) and send them to other platforms. Once the module is installed in your environment, you can create classes that wrap the API calls for you to gather the events you need.
+This module gathers events from the [Puppet Jobs API][1] and the [Activities API][2] using a script executed by a cron job. This data is provided to any available processor scripts during each run. The scripts are provided by any modules that want to make use of this data such as the `puppetlabs-splunk` module. These processor scripts then handle the data they are given and forward it on to their respective platforms.
```
Comment (coreymbe) [inline README.md:18]: Change puppetlabs-splunk to puppetlabs-splunk_hec

Comment (RandomNoun7) [reply README.md:18]: Good catch, thanks!

File: README.md [hunk 2]
```diff
@@ -1,72 +1,298 @@
...
+
+`log_path`
+
+By default the log file will be placed at `/var/log/puppetlabs/pe_event_forwarding/pe_event_forwarding.log`. This path is based on the PE `$settings::logdir` path, but if a value is given to this parameter then the given value will override the default and the logfile will be placed in the directory `"${log_path}/pe_event_forwarding/pe_event_forwarding.log"`. This parameter expects to receive a path to a directory, to which the remainder of the path `/pe_event_forwarding/pe_event_forwarding.log`, will be appended and the required sub directories created.
+
+--------------------------------------------------------------------------------
+
+`lock_path`
+
+By default this file will be placed at `/opt/puppetlabs/pe_event_forwarding/cache/state/events_collection_run.lock`. This path is based on the `$settings::statedir` path, but passing a value to this parameter will override that path and place the lockfile in a custom location on the system. This parameter expects to recieve a path to a directory, to which the remainder of the path, `/pe_event_forwarding/cache/state/events_collection_run.lock`, will be appended and the required sub directories created.
```
Comment (nam054) [inline README.md:264]: *typo recieve --> receive

File: README.md [hunk 3]
```diff
@@ -1,72 +1,298 @@
...
+
+`confdir`
+
+By default this folder will be placed at `/etc/puppetlabs/p_event_forwarding`. This setting is based on the `$settings::confdir` variable, but passing a value to this parameter will override that value and place this directory in a custom location on the system. This parameter expects a path to a directory, inside of which a sub directory `pe_event_forwarding` will be created to hold all of the required files to run events collection. See [Resources Placed On The Machine](#resources-placed-on-the-machine) for details on the files placed in this directory.
+
+--------------------------------------------------------------------------------
+
+`api_window_size`
+
+The size of the pages to fetch from the PE API. All un processed events will still be gathered from the API on each collection cycle, but this parameter allows administrators to control the size of the individual requests to the PE API, to prevent too many events from being returned in a single API call.
```
Comment (nam054) [inline README.md:276]: un processed --> unprocessed

File: README.md [hunk 4]
```diff
@@ -1,72 +1,298 @@
...
+5. [Installing A Processor](#installing-a-processor)
+6. [Token vs Username Password Authentication](#token-vs-username-password-authentication)
+7. [Resources Placed On The Machine](#resources-placed-on-the-machine)
+8. [Logging](#logging)
+9. [Advanced Configuration Options](#advanced-configuration-options)
 
 ## Intro
 
-The common events library is a module makes it easier to gather events from [Puppet API end points](https://puppet.com/docs/pe/2021.2/orchestrator_api_jobs_endpoint.html#orchestrator_api_jobs_endpoint) and send them to other platforms. Once the module is installed in your environment, you can create classes that wrap the API calls for you to gather the events you need.
+This module gathers events from the [Puppet Jobs API][1] and the [Activities API][2] using a script executed by a cron job. This data is provided to any available processor scripts during each run. The scripts are provided by any modules that want to make use of this data such as the [puppetlabs-splunk_hec][3] module. These processor scripts then handle the data they are given and forward it on to their respective platforms.
```
Comment (gsparks) [inline README.md:18]: We may want to make sure that this style of linking will work on the forge. Just a note to check once it's released.

File: README.md [hunk 5]
```diff
@@ -1,72 +1,298 @@
...
+
+--------------------------------------------------------------------------------
+
+`disabled`
+
+Setting this variable to true will events collection from executing. No further events will gathered or sent by any processor until the parameter is removed or set to `true`. It will do this by removing the cron job that executes the events collection script. This is intended for use during maintenance windows of either PE or the recieving platforms, or for any other reason an administrator might want to stop sending events. Simply removing this class from a node configuration would not be enough to stop sending events.
```
Comment (gsparks) [inline README.md:244]: Add word 'prevent' after 'Setting this variable to true will'.

Comment (gsparks) [reply README.md:244]: 'receiving' spelled wrong.

Comment (RandomNoun7) [reply README.md]: Good catch.

File: README.md (hunk 4 above)
Comment (RandomNoun7) [reply README.md:18]: Yeah, we can keep an eye on this and fix if it's broken.

File: README.md [hunk 6]
```diff
@@ -1,72 +1,298 @@
...
+
+`confdir`
+
+By default this folder will be placed at `/etc/puppetlabs/p_event_forwarding`. This setting is based on the `$settings::confdir` variable, but passing a value to this parameter will override that value and place this directory in a custom location on the system. This parameter expects a path to a directory, inside of which a sub directory `pe_event_forwarding` will be created to hold all of the required files to run events collection. See [Resources Placed On The Machine](#resources-placed-on-the-machine) for details on the files placed in this directory.
+
+--------------------------------------------------------------------------------
+
+`api_window_size`
+
+The size of the pages to fetch from the PE API. All unprocessed events will still be gathered from the API on each collection cycle, but this parameter allows administrators to control the size of the individual requests to the PE API, to prevent too many events from being returned in a single API call.
```
Comment (gsparks) [inline README.md:276]: Remove comma after API

**Review by gsparks — APPROVED** [review]

**Pull request merged**